metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatApplicationEventMulticaster.java

```diff
@@ -54,27 +54,23 @@ public MetacatApplicationEventMulticaster(
     }
 
     /**
-     * Post event asynchronously.
+     * Post event. Events will be handles synchronously or asynchronously based on the listener annotation.
      *
      * @param event event
      */
-    public void postAsync(final ApplicationEvent event) {
-        this.asyncEventMulticaster.multicastEvent(event);
-    }
-
-    /**
-     * Post event synchronously.
-     *
-     * @param event event
-     */
-    public void postSync(final ApplicationEvent event) {
+    public void post(final ApplicationEvent event) {
         super.multicastEvent(event);
+        this.asyncEventMulticaster.multicastEvent(event);
     }
 
     @Override
     public void addApplicationListener(final ApplicationListener listener) {
-        super.addApplicationListener(listener);
-        asyncEventMulticaster.addApplicationListener(listener);
+        if (listener.getClass().isAnnotationPresent(AsyncListener.class)) {
```

Comment:
So in this instance you're pulling the annotation present anywhere in the class. But what happens if a single class has multiple listeners both synchronous and asynchronous? Can you use reflection here on the method level rather than the class level?

Reply:
Having it at a method level will be an overkill. We can always define a listener class to be asynchronous. If we need any of the methods to be synchronous then we move those into a new class.